### PR TITLE
[Static references] LPS-182379 Remove static references from layout and knowledge-base modules

### PR DESCRIPTION
**Pull request by MrJohn1911**

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:echo

**Comment by MrJohn1911** [comment]
ci:test:lima

**Comment by zhangyan0701** [comment]
**1. Internal Server Error occurs after adding an attachment or inline image in Wiki page:**

Steps:
Go to Site Administration > Content & Data > Wiki, add a new Wiki page into the Main node.
Add a widget page and add a Wiki portlet into the page.
Edit the new created page in Wiki portlet, add an attachment.
Click Publish.

Results:
In internal server error occurs.
And error in portal console:
`2023-04-21 00:25:53.552 ERROR [http-nio-8080-exec-9][StatusDisplayContext:92] Error: com.liferay.portal.kernel.portlet.PortletContainerException: javax.servlet.ServletException: java.lang.NoClassDefFoundError: Could not initialize class com.liferay.wiki.web.internal.display.context.DLMimeTypeDisplayContextUtil`

**2. AssertionError in integration test com.liferay.wiki.service.test.WikiPageLocalServiceTest failed:**
[com.liferay.wiki.service.test.WikiPageLocalServiceTest](com.liferay.wiki.service.test.WikiPageLocalServiceTest
)
Full console:
https://testray.liferay.com/reports/production/logs/2023-04/test-1-38/test-portal-acceptance-pullrequest(master)/7409/modules-integration-mysql57-jdk8/0/13/jenkins-console.txt.gz?authuser=0

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:echo

**Comment by zhangyan0701** [comment]
ci:test:wiki

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/apps/wiki/wiki-web/src/main/java/com/liferay/wiki/web/internal/display/context/DLMimeTypeDisplayContextUtil.java [hunk 1]
```diff
@@ -15,43 +15,34 @@
...
  */
-@Component(service = {})
 public class DLMimeTypeDisplayContextUtil {
 
 	public static String getCssClassFileMimeType(FileVersion fileVersion) {
-		return _dlMimeTypeDisplayContextUtil._dlMimeTypeDisplayContext.
-			getCssClassFileMimeType(fileVersion.getMimeType());
-	}
+		DLMimeTypeDisplayContext dlMimeTypeDisplayContext =
+			DLMimeTypeDisplayContextUtil._dlMimeTypeDisplayContextSnapshot.
```
Comment (dantewang) [inline modules/apps/wiki/wiki-web/src/main/java/com/liferay/wiki/web/internal/display/context/DLMimeTypeDisplayContextUtil.java:28]: @MrJohn1911 No need to write `DLMimeTypeDisplayContextUtil._dlMimeTypeDisplayContextSnapshot` here.

Also, the PR title says layout and kb modules, but I see layout and wiki. Did you forget `com.liferay.knowledge.base.web.internal.display.context.DLMimeTypeDisplayContextUtil`?

**Review by dantewang — CHANGES_REQUESTED** [review]

File: modules/apps/wiki/wiki-web/src/main/java/com/liferay/wiki/web/internal/display/context/DLMimeTypeDisplayContextUtil.java (hunk 1 above)
Comment (MrJohn1911) [reply modules/apps/wiki/wiki-web/src/main/java/com/liferay/wiki/web/internal/display/context/DLMimeTypeDisplayContextUtil.java:28]: Oh, I picked the wrong class, I will resend with the right one. the `modules/apps/layout/layout-taglib/src/main/java/com/liferay/layout/taglib/internal/frontend/js/loader/modules/extender/npm/NPMResolverProvider.java` was sent [here](https://github.com/liferay-core-infra/liferay-portal/pull/4349/files#diff-639a894ae0ba4825033dc5ad1e115c792f8b6a7cce462d0370f3e82daf280d88).

**Pull request closed without merging**